mailbox/api/src/test/java/org/apache/james/mailbox/model/UidValidityTest.java

```diff
@@ -38,4 +43,79 @@ void ofShouldReturnValidUidValidity() {
         UidValidity uidValidity = UidValidity.of(expectedValue);
         assertThat(uidValidity.asLong()).isEqualTo(expectedValue);
     }
+
+    @Test
+    void zeroUidValidityShouldBeInvalid() {
+        assertThat(UidValidity.of(0).isValid()).isFalse();
+    }
+
+    @Test
+    void negativeUidValidityShouldBeInvalid() {
+        assertThat(UidValidity.of(-1).isValid()).isFalse();
+    }
+
+    @Test
+    void tooBigUidValidityShouldBeInvalid() {
+        assertThat(UidValidity.of(4294967296L).isValid()).isFalse();
+    }
+
+    @Test
+    void idValidityShouldBeValid() {
+        assertThat(UidValidity.of(42).isValid()).isTrue();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenZeroIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 0L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenNegativeIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> -42L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenUpperBoundExclusiveIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967296L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenUpperBoundInclusiveIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967295L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenHigherThanUpperBoundIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967297L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenLowerThanUpperBoundIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967294L))
+            .doesNotThrowAnyException();
+    }
+
+    // Sampling test
+    @RepeatedTest(10000)
+    void randomShouldGenerateValidValues() {
+        assertThatCode(UidValidity::random)
+            .doesNotThrowAnyException();
+    }
+
+    @Disabled("On average upon generating 1.000.000 UidValidity we notice 125 duplicates")
```

Comment:
What are the implications ? Can you state it in the comment ?
And if it is acceptable maybe the test could be one asserting that we have less than 0.0x % of collisions ?

Reply:
> What are the implications ? Can you state it in the comment ?

IMAP clients might be missing  some mailbox switches.

> And if it is acceptable maybe the test could be one asserting that we have less than 0.0x % of collisions ?

What's the point of testing that IMAP RFC is broken?

Reply:
Are you sure you have to blame the RFC? (from 3501)
```
         3) If the mailbox is deleted and a new mailbox with the
            same name is created at a later date, the server must
            either keep track of unique identifiers from the
            previous instance of the mailbox, or it must assign a
            new UIDVALIDITY value to the new instance of the
            mailbox.  A good UIDVALIDITY value to use in this case
            is a 32-bit representation of the creation date/time of
            the mailbox.  It is alright to use a constant such as
            1, but only if it guaranteed that unique identifiers
            will never be reused, even in the case of a mailbox
            being deleted (or renamed) and a new mailbox by the
            same name created at some future time.
```

Reply:
> IMAP clients might be missing some mailbox switches.

The collision has to happen on a mailbox named like a previous one. I don't see that as a problem given the constraint of 32 bit integer, I can't think of a better solution than random ...

Reply:
> IMAP clients might be missing some mailbox switches.

The collision has to happen on a mailbox named like a previous one. I don't see that as a problem given the constraint of 32 bit integer, I can't think of a better solution than random ...